hammer/library/aws/security_groups.py

```diff
@@ -372,6 +374,29 @@ def __str__(self):
         perms = ", ".join([str(perm) for perm in self.permissions])
         return f"{self.__class__.__name__}(Name={self.name}, Id={self.id}, Permissions=[{perms}])"
 
+    def validate_known_ip_soource(self, source_ip):
+        """
+        
+        :param source_ip: ip address
+        :return: boolean
+        """
+        config = Config()
+        known_ip_sources = config.sg.known_ip_sources
+        source_cidr = ipaddress.ip_network(source_ip)
+
+        for known_ip in known_ip_sources:
+            known_ip_cidr = ipaddress.ip_network(known_ip)
+            if known_ip_cidr == source_cidr:
+                return True
+            elif source_ip.endswith("/32"):
+                for ip in known_ip_cidr:
```

Comment:
It seems `ipaddress` module allows to check if ip belongs to network. We can do the check if we know source_cidr is `/32`:
`if source_cidr[-1] in known_ip_cidr:
    return True
`
to avoid the loop over all addresses in subnet.

Reply:
Updated and tested.